app/src/main/java/com/iic/shopingo/ui/request_flow/activities/RequestStateActivity.java

```diff
@@ -18,23 +20,24 @@
       R.layout.activity_request_state_declined
   };
 
-  private SelectShopperActivity.SelectShopperAdapter.ShopRequest request;
+  private OutgoingRequest request;
 
   @Override
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
     request = getIntent().getParcelableExtra(EXTRAS_REQUEST_KEY);
-    setContentView(LAYOUTS[request.status.ordinal()]);
+    setContentView(LAYOUTS[request.getStatus().ordinal()]);
     ButterKnife.inject(this);
   }
 
   @Optional
   @OnClick(R.id.request_state_cancel_button)
   public void onCancelRequest(View view) {
+    request.setStatus(BaseRequest.RequestStatus.CANCELED);
     // TODO: Cancel request in server
-    request.shopper = null;
+    OutgoingRequest newRequest = new OutgoingRequest(request.getShoppingList());
     Intent intent = new Intent(this, SelectShopperActivity.class);
-    intent.putExtra(SelectShopperActivity.EXTRAS_REQUEST_KEY, request);
+    intent.putExtra(SelectShopperActivity.EXTRAS_REQUEST_KEY, newRequest);
```

Comment:
I think it's not relevant to save between sessions, it's more of a real-time thing, no?
I kinda like the flow where you create a request at the beginning and fill it up as you go along, instead of each activity juggling it's own little piece of data and stitching it all together at the end.
What do you think?